requests_oauthlib/oauth2_session.py

```diff
@@ -538,3 +543,40 @@ def register_compliance_hook(self, hook_type, hook):
                 "Hook type %s is not in %s.", hook_type, self.compliance_hook
             )
         self.compliance_hook[hook_type].add(hook)
+
+    def _raise_for_5xx(self, response):
+        # type: (requests.models.Response) -> None
+        """
+        Raise requests.HTTPError if response is an HTTP 5XX error.
+
+        Just like the existing Response.raise_for_status() but ignores 4XX
+        errors.
+
+        :param response: HTTP response object from requests
+        Raises :class:`requests.exceptions.HTTPError`, if a 5XX error occurred.
+        """
+        http_error_msg = ""
+        if isinstance(response.reason, bytes):
+            # We attempt to decode utf-8 first because some servers
+            # choose to localize their reason strings. If the string
+            # isn't utf-8, we fall back to iso-8859-1 for all other
+            # encodings. (See psf/requests PR #3538)
+            try:
+                reason = response.reason.decode("utf-8")
+            except UnicodeDecodeError:
+                reason = response.reason.decode("iso-8859-1")
+        else:
+            reason = response.reason
+
+        if 400 <= response.status_code < 500:
+            pass  # ignored
+
+        elif 500 <= response.status_code < 600:
+            http_error_msg = "%s Server Error: %s for url: %s" % (
+                response.status_code,
+                reason,
+                response.url,
+            )
+
+        if http_error_msg:
+            raise HTTPError(http_error_msg, response=response)
```

Comment:
You only define `http_error_msg` in the `elif 500 <= response.status_code < 600` block above, so maybe worth moving it up there (and then not explicitly checking for it nor assigning it to a variable and raising directly)?